ibis/expr/rewrites.py

```diff
@@ -278,21 +282,39 @@ def rewrite_filter_input(value):
 
 
 @replace(p.Analytic | p.Reduction)
-def window_wrap_reduction(_, window):
-    # Wrap analytic and reduction functions in a window function. Used in the
-    # value.over() API.
+def window_wrap_reduction(_: ops.Analytic | ops.Reduction, window: LegacyWindowBuilder):
+    # Wrap analytic and reduction functions in a window function, merging their ordering.
+    # Used in the value.over() API.
+    order_by_specified_in_op = ("order_by" in _.argnames) and bool(_.order_by)
+    order_by_specified_in_window = bool(window.orderings)
+    if order_by_specified_in_op and order_by_specified_in_window:
+        raise ExpressionError(
+            'Cannot do eg t.group_by("a").order_by("b").mutate(t.c.first(order_by="d"))'
+            'or t.mutate(t.c.first(order_by="d").over(group_by="a", order_by="b"))'
+            'because the "order_by" is specified twice, and is ambiguous.'
+            "Only specify it once, in either place."
+        )
```

Comment:
I could be convinced to make the op ordering take precedence over the window ordering, since it is on the more specific value. eg you might do `t.group_by("a").order_by("b").mutate(c=t.c.first(order_by="d"), e=t.e.collect(), f=t.f.last())`, and you want `e` and `f` to be ordered by the default of "b", but to override this default for "c", and order it by "d".

If a user gets this error message they are going to be perplexed, whereas if we just pick a consistent behavior, then they won't see it and we won't make rare uses illegal. eg if a user has to programmatically detect if an order by is given multiple times, that would be awful for them.

Reply:
I think this is the most obvious behavior. Essentially: order by on a mutate inserts the passed keys to all window/aggregates as the inner-most key. This generalizes to work for specified additional keys as well as no keys specified.

Reply:
This has precedent in the `window_merge_frames` rule and has existed since before the present rewrite system.

Reply:
That said, I do think the behavior of "use both keys, but give the inner most key precedent" is probably the way to go.

Let's make **that** change after the 10.6 release, and I will merge this PR with behavior that matches the existing behavior (which is to **always** place window order_by keys first, even in expressions like `t.col.first(order_by="foo").over(order_by="bar")`).

Reply:
See my comment below. I am a strong -1 on merging the order bys together. I am fine with the more specific one taking precedence and totally overriding the rest, but merging opens up a lot of ambiguity.